### ISSUE TITLE
Raw Read Alignment Fails With Exit Code 256

### ISSUE THREAD

**Issue by malonge** [PacificBiosciences/FALCON#454]
All but the very last 2 jobs in 0-rawreads finish to completion. When I look at the stderr file for these two jobs there doesn't appear to be an obvious source for the error. Here is the initial error message:

```
[INFO]tick: 8192, #updatedTasks: 2, sleep_time=1.000000
[ERROR]Task Fred{'URL': 'task://localhost/d_000b_raw_reads'} failed with exit-code=256
[INFO]Failure ('fail'). Joining 'task://localhost/d_000b_raw_reads'...
[CRITICAL]Any exception caught in RefreshTargets() indicates an unrecoverable error.     Shutting down...
/local/workdir/malonge/FC/lib/python2.7/site-packages/pypeflow-0.1.1-    py2.7.egg/pypeflow/controller.py:537: UserWarning:
            "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
            "! Please wait for all threads / processes to terminate !"
            "! Also, maybe use 'ps' or 'qstat' to check all threads,!"
            "! processes and/or jobs are terminated cleanly.        !"
            "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"

  warnings.warn(shutdown_msg)
[WARNING]#tasks=2, #alive=1
[WARNING]Now, #tasks=2, #alive=0
Traceback (most recent call last):
  File "/local/workdir/malonge/FC/bin/fc_run.py", line 4, in <module>
    __import__('pkg_resources').run_script('falcon-    kit==0.7+git.54779b25532a4e1ace230013fe25f5c51a43ddaa', 'fc_run.py')
  File "/local/workdir/malonge/FC/lib/python2.7/site-packages/pkg_resources/__init__.py", line 746, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/local/workdir/malonge/FC/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1501, in run_script
    exec(code, namespace, namespace)
  File "/local/workdir/malonge/FC/lib/python2.7/site-packages/falcon_kit-  0.7+git.54779b25532a4e1ace230013fe25f5c51a43ddaa-py2.7-linux-x86_64.egg/EGG-    INFO/scripts/fc_run.py", line 5, in <module>
    main(sys.argv)
  File "/local/workdir/malonge/FC/lib/python2.7/site-packages/falcon_kit-0.7+git.54779b25532a4e1ace230013fe25f5c51a43ddaa-py2.7-linux-x86_64.egg/falcon_kit/mains/run1.py", line 576, in main
    main1(argv[0], args.config, args.logger)
  File "/local/workdir/malonge/FC/lib/python2.7/site-packages/falcon_kit-0.7+git.54779b25532a4e1ace230013fe25f5c51a43ddaa-py2.7-linux-x86_64.egg/falcon_kit/mains/run1.py", line 353, in main1
    setNumThreadAllowed=PypeProcWatcherWorkflow.setNumThreadAllowed)
  File "/local/workdir/malonge/FC/lib/python2.7/site-packages/falcon_kit-    0.7+git.54779b25532a4e1ace230013fe25f5c51a43ddaa-py2.7-linux-    x86_64.egg/falcon_kit/mains/run1.py", line 431, in run
    wf.refreshTargets(exitOnFailure=exitOnFailure)
  File "/local/workdir/malonge/FC/lib/python2.7/site-packages/pypeflow-0.1.1-   py2.7.egg/pypeflow/controller.py", line 548, in refreshTargets
    raise Exception('Caused by:\n' + tb)
Exception: Caused by:
Traceback (most recent call last):
  File "/local/workdir/malonge/FC/lib/python2.7/site-packages/pypeflow-0.1.1-    py2.7.egg/pypeflow/controller.py", line 523, in refreshTargets
    rtn = self._refreshTargets(task2thread, objs = objs, callback = callback, updateFreq =      updateFreq, exitOnFailure = exitOnFailure)
File "/local/workdir/malonge/FC/lib/python2.7/site-packages/pypeflow-0.1.1-   py2.7.egg/pypeflow/controller.py", line 740, in _refreshTargets
raise TaskFailureError("Counted %d failure(s) with 0 successes so far." %failedJobCount)
TaskFailureError: 'Counted 1 failure(s) with 0 successes so far.'
```

Looking in 0-rawreads/job_000b/pwatcher.dir at the stderr file i get the following:

```
...
raw_reads.5.raw_reads.50.C1: 121,879 all OK
raw_reads.5.raw_reads.50.C2: 120,769 all OK
raw_reads.5.raw_reads.50.C3: 123,648 all OK
raw_reads.5.raw_reads.50.N0: 144,042 all OK
raw_reads.5.raw_reads.50.N1: 141,156 all OK
raw_reads.5.raw_reads.50.N2: 136,626 all OK
raw_reads.5.raw_reads.50.N3: 140,518 all OK
raw_reads.6.raw_reads.50.C0: 120,771 all OK
raw_reads.6.raw_reads.50.C1: 132,030 all OK
raw_reads.6.raw_reads.50.C2: 133,459 all OK
raw_reads.6.raw_reads.50.C3: 137,798 all OK
raw_reads.6.raw_reads.50.N0: 139,676 all OK
raw_reads.6.raw_reads.50.N1: 152,221 all OK
raw_reads.6.raw_reads.50.N2: 154,030 all OK
raw_reads.6.raw_reads.50.N3: 157,501 all OK
raw_reads.7.raw_reads.50.C0: 127,249 all OK
raw_reads.7.raw_reads.50.C1: 130,786 all OK
raw_reads.7.raw_reads.50.C2: 127,198 all OK
raw_reads.7.raw_reads.50.C3: 132,638 all OK
raw_reads.7.raw_reads.50.N0: 146,313 all OK
raw_reads.7.raw_reads.50.N1: 150,107 all OK
raw_reads.7.raw_reads.50.N2: 146,705 all OK
raw_reads.7.raw_reads.50.N3: 153,604 all OK
raw_reads.8.raw_reads.50.C0: 132,220 all OK
raw_reads.8.raw_reads.50.C1: 125,131 all OK
raw_reads.8.raw_reads.50.C2: 124,251 all OK
raw_reads.8.raw_reads.50.C3: 130,400 all OK
raw_reads.8.raw_reads.50.N0: 151,381 all OK
raw_reads.8.raw_reads.50.N1: 142,729 all OK
raw_reads.8.raw_reads.50.N2: 140,498 all OK
raw_reads.8.raw_reads.50.N3: 148,588 all OK
raw_reads.9.raw_reads.50.C0: 136,249 all OK
raw_reads.9.raw_reads.50.C1: 124,851 all OK
raw_reads.9.raw_reads.50.C2: 122,193 all OK
raw_reads.9.raw_reads.50.C3: 132,169 all OK
raw_reads.9.raw_reads.50.N0: 156,018 all OK
raw_reads.9.raw_reads.50.N1: 144,035 all OK
raw_reads.9.raw_reads.50.N2: 140,720 all OK
raw_reads.9.raw_reads.50.N3: 152,458 all OK
touch /local/workdir/malonge/Heritage_FC/0-rawreads/job_000b/job_000b_done.exit
+ touch /local/workdir/malonge/Heritage_FC/0-rawreads/job_000b/job_000b_done.exit
returned: 256
```

I should note that the job_000b.done.exit file is being touched, but not the job_00b.done file. 
Finally, here is my configuration file:

```
[General]
input_fofn = input.fofn
#input_fofn = preads.fofn

input_type = raw
#input_type = preads

length_cutoff = 500

length_cutoff_pr = 6000

job_type = local
jobqueue = your_queue
sge_option_da = 
sge_option_la =
sge_option_pda =
sge_option_pla =
sge_option_fc = 
sge_option_cns =

pa_concurrent_jobs = 14
cns_concurrent_jobs = 14
ovlp_concurrent_jobs = 14

pa_HPCdaligner_option =  -v -dal128 -t16 -e.70 -l1000 -s1000
ovlp_HPCdaligner_option = -v -dal128 -t32 -h60 -e.96 -l500 -s1000

pa_DBsplit_option = -x500 -s200
ovlp_DBsplit_option = -x500 -s200

falcon_sense_option = --output_multi --min_idt 0.70 --min_cov 1 --max_n_read 500 --n_core 6

overlap_filtering_setting = --max_diff 100 --max_cov 100 --min_cov 1 --bestn 10 --n_core 24
```

I am using a machine with 64 cores and ~1 TB memory. It doesn't appear that there should have been a memory issue. Why would the .done.exit file be created but not the .done file?

Is there another place I could look at that might give more information as to what is causing this error?

Thank you

**Comment by rhallPB** [comment]
Is there any more information in the stdout? Does the merging start? You could try to complete the job manually by executing ./job_000b/rj_000b.sub.sh, comment our everything before `LAcheck -v raw_reads *.las`

**Comment by malonge** [comment]
I will attach both stdout and stderr here.
[falcon_job_000b_std.zip](https://github.com/PacificBiosciences/FALCON/files/520580/falcon_job_000b_std.zip)

Please let me know if, after reading stdout, you still recommend manually completing the job.

Thank you

**Comment by rhallPB** [comment]
Yes I would try to manually complete the job.

**Comment by pb-cdunn** [comment]
There is probably at least one non-OK line in `0-rawreads/job_000b/pwatcher.dir/stderr`? I suspect that your `DALIGNER/LAcheck` is old. Gene fixed a bug several weeks ago.

**Comment by malonge** [comment]
Interesting. So I will update DALIGNER and rerun and see how that goes. If needed, I may also try to run the script manually. 

Thank you both for the help.

**Comment by malonge** [comment]
After updating DALIGNER and manually running the jobs, I have successfully moved on to the fc_concensus stage. Thank you for the help.

**Comment by svm-zhang** [comment]
Hello @malonge,

I had a similar problem and have tried to update DALIGNER to its lastest Head but still got the error. Could you please tell me how you update the DALIGNER and what SHA1 you are at now.

Thanks,
Simo

**Comment by malonge** [comment]
Though I can't be positive, I don't believe my updating of DALIGNER is what solved the problem, since I believe I initially installed it after the last LAcheck bug. If my assumption is correct, then it was actually the manual running of the rj_000b.sub.sh that ended up doing the trick for me. I can't say why though.

**Comment by pb-cdunn** [comment]
If you try running **LAcheck** again (manually), I'll bet it passes now. The bug-fix is in **daligner**, not **LAcheck**.

**Comment by resee** [comment]
I have similar problem. In my last run, after the dealigner jobs finished after running for 10 days after that I could see blastx run by falcon for about 15 days. eventually the blastx also finished but i did not see any progress in my falcon assembly job. when I checked the log file there were 4 jobs which did not finish. As answered in this post I manually ran those 4 jobs. The *.done file already had latest time stamp, so did not had to touch it. I renamed the mypwatched directory and restart the fc_run.py. However I got the same error for the same failed jobs. 
How should I move forward after running rj_*.sub.sh mannually? do I need to touch any *.done file in 0-rawreads ? 

I tried couple time for this genome and couldn't fix the problem. Later I started another smaller genome so that it will finish sooner. I am monitoring the jobs more closely this time. I notice the first step falcon jobs are not anymore in the job queue list of my server. I checked the pypeflow.log and I could see there are 22 unsatisfied jobs. I am can still see some process still running by falcon. This time it is 'canu_cor'  adn 'HotChoco'. 

What are these processes ('canu_cor' , 'HotChoco', and blastx) doing on falcon? Is blastx supposed to run that long ( which was about 15 days in my previous run)?


Thanks in advance
